Source/WebKit/GPUProcess/media/RemoteMediaPlayerProxy.cpp

```diff
@@ -1211,6 +1210,12 @@ void RemoteMediaPlayerProxy::playerContentBoxRectChanged(const WebCore::LayoutRe
         m_player->playerContentBoxRectChanged(contentRect);
 }
 
+void RemoteMediaPlayerProxy::setShouldCheckHardwareSupport(bool value)
+{
+    m_player->setShouldCheckHardwareSupport(value);
+    m_shouldCheckHardwareSupport = value;
```

Comment:
This is a bit odd.
It seems to me this would be more easily given as a parameter to MediaPlayer::load CRABS and MSE methods and passed down to MediaPlayerPrivate at that time.

That would allow to remove these getters and setters, but might make a bigger patch.
Could that be done as a follow-up?

Reply:
The mediaplayer construction in the GPU process is rather cumbersome

In the content process the MediaPlayer creates a MediaPlayerPrivate which here will be a MediaPlayerPrivateRemote. So we have to pass the information to the private player. 

The remote player however, talks to the GPU’s MediaPlayerProxy which is a MediaPlayerClient which will create a new MediaPlayerPrivate. 

So while in the CP, the private need the information, in the GPU it’s the MediaPlayerClient and MediaPlayer. 

I re-used the architecture in place, using setters felt like a less intrusive change over what was existing compare to passing parameters.